pkg/argocd/git_test.go

```diff
@@ -215,3 +216,74 @@ images:
 		})
 	}
 }
+
+func Test_updateKustomizeFile(t *testing.T) {
+	makeTmpKustomization := func(t *testing.T, content []byte) string {
+		f, err := os.CreateTemp("", "kustomization-*.yaml")
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = f.Write(content)
+		if err != nil {
+			t.Fatal(err)
+		}
+		f.Close()
+		t.Cleanup(func() {
+			os.Remove(f.Name())
+		})
+		return f.Name()
+	}
+
+	filter, err := imagesFilter(v1alpha1.KustomizeImages{"foo@sha23456"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name        string
+		content     []byte
+		wantContent []byte
```

Comment:
can we keep `content` and `wantContent` as string to avoid multiple conversions?